### ISSUE TITLE
Failure to connect to Ghidra Server -- odd situation

### ISSUE THREAD

**Issue by jnwilson** [NationalSecurityAgency/ghidra#3917]
I have a tweaky issue that may be mine and not yours. I am hoping someone has some idea what might be going on.

On a Mac using VMware Fusion, I have a setting with remnux(linux) Ghidra 10.1.1 client (static IP) on network A, pfsense router routing between networks A and B, debian Ghidra 10.1.1 server (DHCP) on network B.
I can connect with ghidra client to the ghidra server with no problem.

I have ported this to a setting for mutliple students to use in an internet-disconnected ESx setup available via a Netlab NDG server.
The setting there is remnux(linux) Ghidra 10.1.1 client (static IP) on network A, pfsense router from network A to B, pfsense router from network B to C, debian Ghidra 10.1.1 server (static IP) on network C.
I can communicate from the client machine to the server with no problem, but I cannot connect the ghidra client to the ghidra server.

I have attached Wireshark packet grabs filtering for tcp.port == 13100.
They seem to be *virtually* identical (up to packet receipt ordering).
There are a couple of packets whose sizes differ slightly
(packet 57 in failure vs. 57 in success and packet 49 in failure vs. 48 in success).

I haven't tried to figure out how to get at the unencrypted TLS packets (yet).
I'm hoping someone sees immediately what is going on here.

Any ideas?

![Failure](https://user-images.githubusercontent.com/3485117/151064467-2e255827-82f1-41eb-a99a-af147ea6951c.png)
![Success](https://user-images.githubusercontent.com/3485117/151064476-c60e3daa-a84c-479b-820e-97c5e0e7b94c.png)

**Comment by dev747368** [comment]
you should include your server.conf options

**Comment by jnwilson** [comment]
Default server.conf--no changes


![server conf](https://user-images.githubusercontent.com/3485117/151079883-030a5c54-6fd6-4707-9e62-5e481245a802.png)
.

**Comment by mumbel** [comment]
> Allows the base TCP port to be specified (default: 13100). The server utilizes three (3) TCP ports starting with the specified base port (e.g., 13100,13101 and 13102). The ports utilized are logged by the server during startup.  

Did you see any trafffic on the other ports?

**Comment by dev747368** [comment]
Also, in your description of your network, you didn't specify if there were any NATing going on.  When there is NATing, you do need to use configuration options in the server.conf to get it working.

**Comment by jnwilson** [comment]
The client is NATed, the server is not.
The found this about NAT, but it is for server NATed: https://github.com/NationalSecurityAgency/ghidra/issues/414

The quandary is that connections *work* in one setting (network of machines running under VMware Fusion on Mac) and *break* in another (same network of machines running on ESx hosts in Vsphere).

BTW, there is, indeed traffic on 13101 and 13102.
I could not find a way to disable TLS on the ghidra server so am looking at the extra tedium of installing my own cert and decrypting in wireshark now. I could also look at the config file parser to see if a no-tsl option exists, but one can't follow ll paths simultaneously.

**Comment by ghidra1** [comment]
The server will auto-generate its own server cert which should be fine unless you have installed a cacert file for the Ghidra client app.  You should not need to dig into the encrypted packet content to diagnose this issue.

Have you looked at the client side log files for connection error details?

If this is a TLS issue I generally enable the following java property on the client side to obtain detailed logging.  I would not use Wireshark to diagnose such issues beyond simple connectivity which appears to be OK based on your limited trace.

You can temporarily add the following to a client-side `launch.properties` file to obtain the detailed logging:

`VMARGS_LINUX=-Djavax.net.debug=all`

**Comment by jnwilson** [comment]
Sadly, the situation is resolved.
As far as I know, there is *nothing* I did that would resolve the problem.
I was looking at the server.log and loglevel INFO didn't provide much, so I set the wrapper.logfile.loglevel to DEBUG and restarted the server.
Connections from clients started working.

I reset the loglevel to INFO to see if that was, indeed, the solution and, sadly, connections still work.
So I have no idea what else may have changed, but it did.

Thank you for your help. I think we should consider this matter closed by PoTM.
If I ever find out what happened and how it got fixed, I will report back.
